docs/DeveloperGuide.adoc

```diff
@@ -46,177 +39,186 @@ The following class plays an important role at the architecture level:
+via the JavaScript card (Java card to be implemented soon) and
 
-Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.
+the `JavascriptRunner` class
 
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+* can evaluate JS code from a file using the `FileImporter` class
+* can evaluate JS code as an input string
 
-image::UndoRedoState0.png[]
+the `JavaRunner` class
 
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+* Read/write to the `Solution.java` file during runtime after user has typed
+Java code into the file
+* Compile and run `Solution.java`
+* the Java code written in each card is stored in the card itself, but this `Solution.java` will be overwritten
+and used to run the code in each card.
 
-image::UndoRedoState1.png[]
+To be implemented: An in-app Flashcard Pro compiler and debugger for JS and Java as a playground for the user
 
-Step 3. The user executes `add n/David ...` to add a new person. The `add` command also calls `Model#commitAddressBook()`, causing another modified address book state to be saved into the `addressBookStateList`.
 
-image::UndoRedoState2.png[]
+[[Design-Commons]]
+=== Common classes
 
-[NOTE]
-If a command fails its execution, it will not call `Model#commitAddressBook()`, so the address book state will not be saved into the `addressBookStateList`.
+Classes used by multiple components are in the `dream.fcard.datastructures` package and root of `dream.fcard.util`
 
-Step 4. The user now decides that adding the person was a mistake, and decides to undo that action by executing the `undo` command. The `undo` command will call `Model#undoAddressBook()`, which will shift the `currentStatePointer` once to the left, pointing it to the previous address book state, and restores the address book to that state.
+== Implementation
 
-image::UndoRedoState3.png[]
+This section describes some noteworthy details on how certain features are implemented.
 
-[NOTE]
-If the `currentStatePointer` is at index 0, pointing to the initial address book state, then there are no previous address book states to restore. The `undo` command uses `Model#canUndoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the undo.
 
-The following sequence diagram shows how the undo operation works:
+// tag::parsejson[]
+==== Implementation
 
-image::UndoSequenceDiagram.png[]
+The parsing is done by `JsonParser`.
```

Comment:
I'm not really sure about which parsing process does it refer to? Is it going to parse from storage or user input?